=== modified file 'fields.py'
--- fields.py	2007-09-15 20:29:05 +0000
+++ fields.py	2008-01-10 04:48:07 +0000
@@ -26,6 +26,13 @@
 
         # Save tags back to the database post-save
         dispatcher.connect(self._save, signal=signals.post_save, sender=cls)
+        dispatcher.connect(self._post_init, signal=signals.post_init, sender=cls)
+
+    def _post_init(self, sender, instance):
+        if instance._get_pk_val() is None:
+            self._set_instance_tag_cache(instance, '')
+        else:
+            self._set_instance_tag_cache(instance, tags2str(Tag.objects.get_for_object(instance)))
 
     def __get__(self, instance, owner=None):
         """

